File: dotnet_distance_generator/trunk/src/csharp/tests/TopCoder/Web/Distance/AccuracyTests/GeographicalDistanceCalculatorTests.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using TopCoder.Web.Distance.Data;
using TopCoder.Web.Distance.DistanceCalculators;


namespace TopCoder.Web.Distance.AccuracyTests
{
    [TestFixture, CoverageExclude]
    public class GeographicalDistanceCalculatorTests
    {
        /// <summary>
        /// Tests for both defined distances
        /// </summary>
        [Test]
        public void TestBothDefined()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[0]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[0], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.AreEqual((float)(0.256 / 0.257), list[0]);
        }

        /// <summary>
        /// Tests for defined/undefined distances
        /// </summary>
        [Test]
        public void TestDefinedUndefined()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[0]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[0], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.IsTrue(0.0f >= list[2]);
        }


        /// <summary>
        /// Tests for defined/undefined distances
        /// </summary>
        [Test]
        public void TestUndefinedDefined()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[4]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[4], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            //System.Console.WriteLine(AccuracyTestHelper
            //       .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[4]).GeographicalDistance);
            Assert.IsTrue(0.0f >= list[0]);
        }

        /// <summary>
        /// Tests for both undefined distances
        /// </summary>
        [Test]
        public void TestBothUndefined()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[4]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[4], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.IsTrue(0.0f >= list[2]);
        }

        /// <summary>
        /// Tests for zero max distance
        /// </summary>
        [Test]
        public void TestZeroMaxDistance()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(
                AccuracyTestHelper.GetTestDataAccess().GetMember(156859),
                AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(156859,
                AccuracyTestHelper.GetAllEnum()),
                AccuracyTestHelper.GetAlgoritmEnum());
            Assert.AreEqual(0.0f, list[1]);
            Assert.IsTrue(0.0f >= list[1]);
        }

        /// <summary>
        /// Tests for undef max distance
        /// </summary>
        [Test]
        public void TestUndefMaxDistance()
        {
            IDistanceCalculator calculator = new GeographicalDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(156860),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            156860, AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.IsTrue(0.0f >= list[1]);
        }

    }
}

File: java_distance_generator/trunk/src/java/tests/com/topcoder/web/tc/distance/accuracytests/GeographicalDistanceCalculatorTests.java
package com.topcoder.web.tc.distance.accuracytests;

import java.util.List;

import junit.framework.TestCase;

import com.topcoder.web.tc.distance.DistanceCalculator;
import com.topcoder.web.tc.distance.calculators.GeographicalDistanceCalculator;
import com.topcoder.web.tc.distance.data.Member;

/**
 * Accuracy test cases for GeographicalDistanceCalculator
 *
 * @author aksonov
 *
 */
public class GeographicalDistanceCalculatorTests extends TestCase {

    /**
     * Tests for both defined distances
     */
    public void testBothDefined() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Double> list = calculator.getDistance(TestHelper
                .getTestDataAccess().getMember(TestHelper.MEMBER_IDS[0]),
                TestHelper.getTestDataAccess().getRelatedMembers(
                        TestHelper.MEMBER_IDS[0], null), TestHelper
                        .getAlgoritmEnum());

        assertEquals(0.256/0.257, list.get(0));
    }

    /**
     * Tests for defined/undefined distances
     */
    public void testDefinedUndefined() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Double> list = calculator.getDistance(TestHelper
                .getTestDataAccess().getMember(TestHelper.MEMBER_IDS[0]),
                TestHelper.getTestDataAccess().getRelatedMembers(
                        TestHelper.MEMBER_IDS[0], null), TestHelper
                        .getAlgoritmEnum());


        assertEquals(-1.0, list.get(2));
    }

    /**
     * Tests for defined/undefined distances
     */
    public void testUndefinedDefined() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Double> list = calculator.getDistance(TestHelper
                .getTestDataAccess().getMember(TestHelper.MEMBER_IDS[4]),
                TestHelper.getTestDataAccess().getRelatedMembers(
                        TestHelper.MEMBER_IDS[4], null), TestHelper
                        .getAlgoritmEnum());


        assertEquals(-1.0, list.get(2));

    }

    /**
     * Tests for both undefined distances
     */
    public void testBothUndefined() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Double> list = calculator.getDistance(TestHelper
                .getTestDataAccess().getMember(TestHelper.MEMBER_IDS[4]),
                TestHelper.getTestDataAccess().getRelatedMembers(
                        TestHelper.MEMBER_IDS[4], null), TestHelper
                        .getAlgoritmEnum());


        assertEquals(-1.0, list.get(0));
    }

    /**
     * Tests for zero max distance
     */
    public void testZeroMaxDistance() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Member> members = TestHelper.getTestDataAccess()
                .getRelatedMembers(156859, TestHelper.getAlgoritmDevEnum());

        List<Double> list = calculator.getDistance(members.get(0), members,
                TestHelper.getAlgoritmEnum());

        assertEquals(0.0, list.get(1));
    }

    /**
     * Tests for undef max distance
     */
    public void testUndefMaxDistance() {
        DistanceCalculator calculator = new GeographicalDistanceCalculator();
        List<Member> members = TestHelper.getTestDataAccess()
                .getRelatedMembers(156860, TestHelper.getAlgoritmDevEnum());

        List<Double> list = calculator.getDistance(members.get(0), members,
                TestHelper.getAlgoritmEnum());

        assertEquals(-1.0, list.get(1));
    }

}
